src/experimental/tools/videoThumbnailLoader/init_source_buffer.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EMPTY,
+  Observable,
+  of as observableOf,
+} from "rxjs";
+import {
+  catchError,
+  filter,
+  map,
+  mapTo,
+  mergeMap,
+  shareReplay,
+} from "rxjs/operators";
+import createSegmentLoader from "../../../core/pipelines/segment/create_segment_loader";
+import { QueuedSourceBuffer } from "../../../core/source_buffers";
+import dash from "../../../transports/dash";
+import prepareSourceBuffer from "./prepare_source_buffer";
+import { IContentInfos } from "./types";
+
+const _currentVideoSourceBuffers: WeakMap<HTMLMediaElement,
+                                          QueuedSourceBuffer<Uint8Array>> = new WeakMap();
+const _currentContentInfos: WeakMap<HTMLMediaElement,
+                                    IContentInfos> = new WeakMap();
+
+const { loader, parser } = dash({ lowLatencyMode: false }).video;
+const segmentLoader = createSegmentLoader(
+  loader,
+  { maxRetry: 0,
+    maxRetryOffline: 0,
+    initialBackoffDelay: 0,
+    maximumBackoffDelay: 0, });
+
+/**
+ * Get current source buffer :
+ * - If already created for current representation, reuse
+ * - If new codecs and/or init URL, create a new one
```

Comment:
We cannot create a new SourceBuffer while one was already active on the current MediaSource. So this comment does not make may sense to me.

Also why should we need to create a new SourceBuffer for a new init segment (also we're not talking about "URL" anymore) ?

Even for different codecs, we might just re-use the same SourceBuffer and call the `changeType` method.